### PR TITLE
port client and trainer service tests from vibecoders

### LINKED ISSUE #344: Bring VibeCoders ClientServiceTests to main project
Port ClientServiceTests.cs from VibeCoders.Tests/Unit/Services/ into the Tests project. Fix or rewrite tests that don't compile against current services. Focus on branch coverage.

### LINKED ISSUE #345: Bring VibeCoders TrainerServiceTests to main project
Port TrainerServiceTests.cs from VibeCoders.Tests/Unit/Services/ into the Tests project. Fix or rewrite tests that don't compile against current services. Focus on branch coverage.

### PR DESCRIPTION
**Pull request by AndaChiorean**
Ported the ClientService and TrainerService tests from VibeCoders. Had to rewrite them pretty heavily since the API changed a lot — the main project uses DTOs everywhere and async methods, while the original tests were sync with direct model objects.

ClientServiceTests (17 tests) covers workout history retrieval, finalize/modify workout flows, nutrition plan lookup, profile snapshots, dashboard delegation, and the sync edge cases.

TrainerServiceTests (15 tests) covers feedback saving, routine assignment (new + editing), nutrition plan assignment/creation, template CRUD, and client listing.

All 32 tests green.

Closes #344
Closes #345

### CODE REVIEW

File: Tests/Services/TrainerServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+                new() { Name = "Weird Machine", MuscleGroup = "ALIEN_MUSCLE" },
+            },
+        };
+
+        var service = this.CreateService();
+        await service.AssignNewRoutineAsync(request);
+
+        this.trainerRepository.Verify(r => r.SaveTrainerWorkoutAsync(
+            It.Is<WorkoutTemplate>(t => t.Exercises.First().MuscleGroup == MuscleGroup.OTHER)),
+            Times.Once);
```
Comment (razvanspartan) [inline Tests/Services/TrainerServiceTests.cs:407]: again, try to remove useless tests.. 800 lines of tests?? seems a bit much

**Review by razvanspartan — COMMENTED** [review]
try to remove more tests

File: Tests/Services/TrainerServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+        var service = this.CreateService();
+        var names = await service.GetAllExerciseNamesAsync();
+
+        Assert.Equal(3, names.Count);
+        Assert.Contains("Bench Press", names);
+        Assert.Contains("Rows", names);
+    }
+
+    [Fact]
+    public async Task SaveTrainerWorkout_maps_unknown_type_to_TRAINER_ASSIGNED()
```
Comment (razvanspartan) [inline Tests/Services/TrainerServiceTests.cs:370]: also wrong names

**Review by razvanspartan — CHANGES_REQUESTED** [review]

**Review by razvanspartan — APPROVED** [review]
